src/coreclr/jit/codegenxarch.cpp

```diff
@@ -1303,6 +1303,55 @@ void CodeGen::genCodeForCompare(GenTreeOp* tree)
     }
 }
 
+//------------------------------------------------------------------------
+// genCodeForCompare: Produce code for a GT_SELECT/GT_SELECT_HI node.
+//
+// Arguments:
+//    select - the node
+//
+void CodeGen::genCodeForSelect(GenTreeOp* select)
```

Comment:
Preexisting, no change required here: wait, so `GT_SELECT` is a descendant of `GenTreeOp`, but the added `gtCond` operand comes first in the operand order? That does not seem like an exception worth making...

Reply:
Yes, it's a little odd. Probably we should just rename it to `GenTreeSelect` and have its operands inline. Or add `GenTreeTernary` and shift the operands.

Reply:
So, 'GenTreeTernary' with condition as op1, true path as op2 and false path as op3? That would probably make more sense, but it'll be annoying to fix them all up

Reply:
Yes, that would be the idea. Alternatively we make `GenTreeTernary` -> `GenTreeSelect` with `gtCond`, `gtTrueVal` and `gtFalseVal` fields.